cl/corpus_importer/management/commands/scrape_pacer_free_opinions.py

```diff
@@ -231,23 +233,43 @@ def get_and_save_free_document_reports(
                 continue
             dates = make_date_range_tuples(date_start, date_end, gap=7)
 
-        # Iterate through the gap in dates either short or long
-        for _start, _end in dates:
-            exc = fetch_doc_report(
-                pacer_court_id, _start, _end  # type: ignore
-            )
-            if exc:
-                # Something happened with the queried date range, abort process for
-                # that court
-                court_failed = True
-                break
+            # Iterate through the gap in dates either short or long
+            for _start, _end in dates:
+                exc = fetch_doc_report(
+                    pacer_court_id, _start, _end  # type: ignore
+                )
+                if exc:
+                    # Something happened with the queried date range, abort process for
+                    # that court
+                    court_failed = True
+                    break
 
-            # Wait 1s between queries to try to avoid a possible throttling/blocking
-            # from the court
-            time.sleep(1)
+                # Wait 1s between queries to try to avoid a possible throttling/blocking
+                # from the court
+                time.sleep(1)
 
-        if court_failed:
-            continue
+            if court_failed:
+                continue
+    else:
+        # Custom date range, alternate courts on a weekly basis to generate report
+        # when running sweep based on specified date range
+        for _start, _end in dates:
+            court_failed = False
+            for pacer_court_id in pacer_court_ids:
+                exc = fetch_doc_report(
+                    pacer_court_id, _start, _end  # type: ignore
+                )
+                if exc:
+                    # Something happened with the queried date range, abort process for
+                    # that court
+                    court_failed = True
+                    break
+                # Wait 1s between queries to try to avoid a possible throttling/blocking
+                # from the court
+                time.sleep(1)
```

Comment:
OK, so the way this works is that if the user provides dates we iterate their courts for each date range. But if the user doesn't provide dates, we iterate the dates for each court, right? 

Three thoughts:
1. I don't understand why we need different logic for whether not dates were provided. 
2. If we do need different logic because one is better, why not apply it to both branches of the `if` statement?
3. I think this part was working fine before, and it has 1s waits between each iteration, so why do we need any changes here at all? The big concern is that we'd hammer a server with tons of requests. Yes, it's better to iterate courts for dates instead of dates for courts, but I think one request at a time with a 1s delay was fine? Or were there errors?

Reply:
1. When days are not provided, it is the daily cron and applied the end date as today and the start as today minus 10 days, but below I already commented how the cron would be so as not to have to do that

2. In one part of the process, the end date is obtained based on a table in the database. This date is the last day that the court was consulted. In this way, we do not lose anything, since the command will make the necessary requests until the court documents are updated to today. It is good to keep this like this because if for some reason there is a problem with the court server and we do not get anything, the next day we will get everything from the last day that everything worked correctly. For the sweep, we cannot use this table because it keeps telling us if we are up to date or not. That is why there are two "if" statements with different logic, one for the daily cron and another for the sweep.

3. I think the previous process was correct for the daily cron since there are few requests, for the full sweep many requests will be made per court (from 1950 until this month), that's why I thought that for the sweep it would be better to iterate weekly for each court instead of processing the entire court, if you think 1 second is enough I can revert it back to before without problem, how common is it that we get blocked for making too many requests?